pallets/identity/src/keys.rs

```diff
@@ -678,39 +684,30 @@ impl<T: Config> Module<T> {
         id: IdentityId,
         secondary_keys: Vec<SecondaryKey<T::AccountId>>,
     ) {
-        <Module<T>>::link_account_key_to_did(&primary_key, id);
+        // Link primary key.
+        <Module<T>>::add_key_record(&primary_key, KeyRecord::PrimaryKey(id));
+        // Link secondary keys.
         for sk in &secondary_keys {
-            if let Signatory::Account(key) = &sk.signer {
-                Self::link_account_key_to_did(key, id);
-            }
+            Self::add_key_record(&sk.key, KeyRecord::SecondaryKey(id, sk.permissions.clone()));
         }
 
-        let record = DidRecord {
-            primary_key: primary_key.clone(),
-            secondary_keys,
-            ..Default::default()
-        };
-        <DidRecords<T>>::insert(&id, record);
-
         Self::deposit_event(RawEvent::DidCreated(id, primary_key, vec![]));
```

Comment:
Pre-existing, but looks like `secondary_keys` is not embedded in the event.
If that's what we want, you can consume `secondary_keys` before and avoid `sk.permissions.clone()`.